peps/pep-0750.rst

```diff
@@ -0,0 +1,895 @@
+may be tempted to use an f-string to embed values into their SQL expression instead of
+using the ``?`` placeholder and passing the values as a tuple to avoid an
+`SQL injection attack <https://en.wikipedia.org/wiki/SQL_injection>`__.
+
+Tag strings address both these problems by extending the f-string syntax to provide
+developers access to the string and its interpolated values before they are combined. In
+doing so, tag strings may be interpreted in many different ways, opening up the
+possibility for DSLs and other custom string processing.
+
+Proposal
+========
+
+This PEP proposes customizable prefixes for f-strings. These f-strings then
+become a "tag string": an f-string with a "tag function." The tag function is
+a callable which is given a sequence of arguments for the parsed tokens in
+the string.
+
+Here's a very simple example. Imagine we want a certain kind of string with
+some custom business policies: uppercase the value and add an exclamation point.
+
+Let's start with a tag string which simply returns a static greeting:
+
+.. code-block:: python
+
+    def greet():
+        """Give a static greeting."""
+        return "Hello!"
+
+    assert greet"Hi" == "Hello!"  # Use the custom "tag" on the string
+
+As you can see, ``greet`` is just a callable, in the place that the ``f``
+prefix would go. Let's look at the args:
+
+.. code-block:: python
+
+    def greet(*args):
+        """Uppercase and add exclamation."""
+        salutation = args[0].upper()
+        return f"{salutation}!"
+
+    greeting = greet"Hello"  # Use the custom "tag" on the string
+    assert greeting == "HELLO!"
+
+The tag function is passed a sequence of arguments. Since our tag string is simply
+``"Hello"``, the ``args`` sequence only contains a string-like value of ``'Hello'``.
+
+With this in place, let's introduce an *interpolation*. That is, a place where
+a value should be inserted:
+
+.. code-block:: python
+
+    def greet(*args):
+        """Handle an interpolation."""
+        # The first arg is the string-like value "Hello " with a space
+        salutation = args[0].strip()
+        # The second arg is an "interpolation"
+        interpolation = args[1]
+        # Interpolations are tuples, the first item is a lambda
+        getvalue = interpolation[0]
+        # It gets called in the scope where it was defined, so
+        # the interpolation returns "World"
+        result = getvalue()
+        recipient = result.upper()
+        return f"{salutation} {recipient}!"
+
+    name = "World"
+    greeting = greet"Hello {name}"
+    assert greeting == "Hello WORLD!"
+
+The f-string interpolation of ``{name}`` leads to the new machinery in tag
+strings:
+
+- ``args[0]`` is still the string-like ``'Hello '``, this time with a trailing space
+- ``args[1]`` is an expression -- the ``{name}`` part
+- Tag strings represent this part as an *interpolation* object
+- An interpolation is a tuple whose first item is a lambda
```

Comment:
Here you say tuple and below tuple like. 

Personally I would just defer the 3rd and 4th point to where you fully describe interpolations.

Reply:
Good points, we'll make bullet 3 that just says discussed below.

Reply:
Merged these changes from @pauleveritt